examples/src/main/kotlin/org/dashevo/examples/GetNetworkStatus.kt

```diff
@@ -0,0 +1,144 @@
+/**
+ * Copyright (c) 2020-present, Dash Core Group
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+package org.dashevo.examples
+
+import io.grpc.Status
+import io.grpc.StatusRuntimeException
+import org.dashevo.Client
+import org.dashevo.dapiclient.DapiClient
+import org.dashevo.dpp.identifier.Identifier
+import org.dashevo.platform.Platform
+
+class GetNetworkStatus {
+
+    companion object {
+
+        private const val identityId = "BSNxbs99zCFvEtK8qrewYTVxNKt4DnFQ8sPbsb8nrDuf"
+        private val identifier = Identifier.from(identityId)
+
+        lateinit var sdk: Client
+        lateinit var platform: Platform
+
+        @JvmStatic
+        fun main(args: Array<String>) {
+            if (args.isEmpty()) {
+                println("Usage: GetStatus network")
+                return
+            }
+            sdk = Client(args[0])
+            platform = sdk.platform
+
+            start()
+        }
+
+        private fun getMnList(): List<Map<String, Any>> {
+            val success = 0
+            do {
+                try {
+                    val baseBlockHash = platform.client.getBlockHash(0)
+                    val blockHash = platform.client.getBestBlockHash()
+
+                    val mnListDiff = platform.client.getMnListDiff(baseBlockHash!!, blockHash!!)
+                    return mnListDiff!!["mnList"] as List<Map<String, Any>>
+                } catch (e: Exception) {
+                    println("Error: $e")
+                }
+            } while (success == 0)
+            return listOf()
+        }
+
+        private fun start() {
+            // Get full masternode list
+            val mnList = getMnList()
+
+            println("${mnList.size} masternodes found")
+            println("Ignoring ${mnList.filter { it["isValid"] == false }.size} invalid masternodes")
+            println("Checking RPC connections for ${mnList.filter { it["isValid"] == true }.size} valid masternodes\n****************")
+
+            val badNodes = arrayListOf<Map<String, Any>>()
+            val invalidNodes = arrayListOf<String>()
+            val goodNodes = arrayListOf<String>()
+            for (mn in mnList) {
+                val service = (mn["service"] as String).split(":")[0]
+
+                if (mn["isValid"] == false) {
```

Comment:
FYI, I refactored a little if you're interested. This check is actually unnecessary if you just make the for loop iterate over only the valid nodes.
https://repl.it/@thephez/Testnet-MN-Comm-Check#index.js